src/Illuminate/Support/Testing/Fakes/ExceptionHandlerFake.php

```diff
@@ -0,0 +1,267 @@
+     */
+    public function assertNothingReported()
+    {
+        Assert::assertEmpty(
+            $this->reported,
+            sprintf(
+                'The following exceptions were reported: %s.',
+                implode(', ', array_map('get_class', $this->reported)),
+            ),
+        );
+    }
+
+    /**
+     * Assert if an exception of the given type has been reported.
+     *
+     * @param  \Closure|string  $exception
+     * @return void
+     */
+    public function assertReported(Closure|string $exception)
+    {
+        $message = sprintf(
+            'The expected [%s] exception was not reported.',
+            is_string($exception) ? $exception : $this->firstClosureParameterType($exception)
+        );
+
+        if (is_string($exception)) {
+            Assert::assertTrue(
+                in_array($exception, array_map('get_class', $this->reported), true),
+                $message,
+            );
+
+            return;
+        }
+
+        Assert::assertTrue(
+            collect($this->reported)->contains(
+                fn (Throwable $e) => $this->firstClosureParameterType($exception) === get_class($e)
+                    && $exception($e) === true,
+            ), $message,
+        );
+    }
+
+    /**
+     * Assert the number of exceptions that have been reported.
+     *
+     * @param  int  $count
+     * @return void
+     */
+    public function assertReportedCount(int $count)
+    {
+        $total = collect($this->reported)->count();
+
+        PHPUnit::assertSame(
+            $count, $total,
+            "The total number of exceptions reported was {$total} instead of {$count}."
+        );
+    }
+
+    /**
+     * Assert if an exception of the given type has not been reported.
+     *
+     * @param  \Closure|string  $exception
+     * @return void
+     */
+    public function assertNotReported(Closure|string $exception)
+    {
+        try {
+            $this->assertReported($exception);
+        } catch (ExpectationFailedException $e) {
+            return;
+        }
+
+        throw new ExpectationFailedException(sprintf(
+            'The expected [%s] exception was not reported.',
+            is_string($exception) ? $exception : $this->firstClosureParameterType($exception)
+        ));
+    }
+
+    /**
+     * Report or log an exception.
+     *
+     * @param  \Throwable  $e
+     * @return void
+     */
+    public function report($e)
+    {
+        if ($this->shouldReport($e) && $this->isFakedException($e)) {
+            $this->reported[] = $e;
+
+            if ($this->throwOnReport) {
+                throw $e;
+            }
+        }
+
+        $this->handler->report($e);
+    }
+
+    /**
+     * Determine if the exception should be reported.
+     *
+     * @param  \Throwable  $e
+     * @return bool
+     */
+    public function shouldReport($e)
+    {
+        return $this->runningWithoutExceptionHandling() || $this->handler->shouldReport($e);
+    }
```

Comment:
Do we need to the call `runningWithoutExceptionHandling` here?

The fake's constructor types the `ExceptionHandler` class, which has a `shouldReport` method on it.

Any handler, anonymous or not, will tell us if they want exceptions reported. The handler bound by the `withoutExceptionHandling` test method does this:

https://github.com/laravel/framework/blob/7c0e6ad5619675b280ba3963a47ee3117df1ba16/src/Illuminate/Foundation/Testing/Concerns/InteractsWithExceptionHandling.php#L100-L109